platform-hub-api/app/services/kubernetes/static_token_service.rb

```diff
@@ -0,0 +1,82 @@
+module Kubernetes
+  module StaticTokenService
+    extend self
+
+    # Static tokens are stored in HashRecord as:
+    # data: [
+    #   {"token" => "token1", user" => "user1", "uid" => "uid1", "groups" => ["group1","group2"]}
+    #   {"token" => "token2", user" => "user2", "uid" => "uid2", "groups" => []}
+    #   ...
+    # ]
+    # Note: All tokens are encrypted!
+
+    def create_or_update(cluster, kind, user_name, groups = [])
+      static_tokens = get_static_tokens(cluster, kind)
+
+      record = static_tokens.data.find do |t|
+        t['user'] == user_name.to_s
+      end
+
+      if record.nil?
+        token = Kubernetes::TokenService.generate_secure_random
+        record = Hashie::Mash.new(
+          token: ENCRYPTOR.encrypt(token),
+          user: user_name,
+          uid: Kubernetes::TokenService.generate_secure_random,
+          groups: Kubernetes::TokenService.send(:cleanup, groups)
+        )
+
+        static_tokens.data << record
+      else
+        record['groups'] = groups.present? ? 
+          Kubernetes::TokenService.send(:cleanup, groups) : []
+      end
+
+      static_tokens.save!
+      "Created/updated #{kind} account for `#{user_name}` (token: #{ENCRYPTOR.decrypt(record['token'])})"
```

Comment:
Totally agree. I'll review of the stdout logging in this piece and make sure no user / sensitive data is ever there. We can always refer to Audit events if we need more details.